### PR TITLE
Disable User Registration

### PR DESCRIPTION
**Pull request by calvinlu3**
This is related to the [disable registration ticket](https://github.com/oncokb/oncokb/issues/2792)

#### Changes
- Created a `DisableEndpoint`  aop annotation that will send back `HttpStatus.Forbidden` to denote that the endpoint is disabled. 
   - Reference: https://www.baeldung.com/spring-aop-annotation
- Show an alert message on frontend
- Added property to set application to readonly mode.

### CODE REVIEW

File: src/main/java/org/mskcc/cbio/oncokb/aop/api/DisableEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package org.mskcc.cbio.oncokb.aop.api;
+
+import java.lang.annotation.Target;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface DisableEndpoint {}
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/aop/api/DisableEndpoint.java:10]: What do you think if we call it `APIConditionallyEnabled`?

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java [hunk 2]
```diff
@@ -105,6 +106,7 @@ public AccountResource(UserRepository userRepository, UserService userService,
      * @throws EmailAlreadyUsedException {@code 400 (Bad Request)} if the email is already used.
      * @throws LoginAlreadyUsedException {@code 400 (Bad Request)} if the login is already used.
      */
+    @DisableEndpoint
```
Comment (zhx828) [inline src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java:109]: The reason I proposed the new name is at the place using this annotation, it's not so clear what it is for.

**Review by zhx828 — COMMENTED** [review]
Just a minor comment.

File: src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java (hunk 2 above)
Comment (calvinlu3) [reply src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java:109]: I agree that the name makes it clearer. Will make that change.

**Pull request merged**